bugsnag-android-core/src/main/java/com/bugsnag/android/SessionFilenameInfo.kt

```diff
@@ -0,0 +1,55 @@
+package com.bugsnag.android
+
+import java.io.File
+import java.util.UUID
+
+/**
+ * Represents important information about a session filename.
+ * Currently the following information is encoded:
+ *
+ * uuid - to disambiguate stored error reports
+ * timestamp - to sort error reports by time of capture
+ */
+internal data class SessionFilenameInfo(
+    val timestamp: Long,
+    val uuid: String,
+) {
+
+    fun encode(): String {
+        return toFilename(timestamp, uuid)
+    }
+
+    internal companion object {
+
+        /**
+         * Generates a filename for the session in the format
+         * "[UUID][timestamp]_v2.json"
+         */
+        fun toFilename(timestamp: Long, uuid: String): String {
+            return "${uuid}${timestamp}_v2.json"
+        }
+
+        fun defaultFilename(): String {
+            return toFilename(System.currentTimeMillis(), UUID.randomUUID().toString())
+        }
+
+        fun fromFile(file: File): SessionFilenameInfo {
+            return SessionFilenameInfo(
+                findTimestampInFilename(file),
+                findUuidInFilename(file)
+            )
+        }
+
+        private fun findUuidInFilename(file: File): String {
+            return file.name.substring(0, 35)
+        }
+
+        fun findTimestampInFilename(file: File): Long {
+            try {
+                return java.lang.Long.parseLong(file.name.substring(file.name.indexOf("_")))
```

Comment:
I might be missing something, but it doesn't look like this took the UUID into account?

```suggestion
            return file.name.substring(36, file.name.indexOf("_")).toLongOrNull() ?: 0
```